### PR TITLE
GO: make the alert messages of taint-tracking queries more consistent

### PR DESCRIPTION
**Pull request by erik-krogh**
I just did this for [Ruby](https://github.com/github/codeql/pull/10304), [Python](https://github.com/github/codeql/pull/10252), and [JavaScript](https://github.com/github/codeql/pull/10286).   

I'm rewriting alert messages for the taint tracking queries:

- I tried to fit the message to the template: `This path depends on a user-provided value.` 
- If that didn't work I tried `User-provided value flows to this location and is used in a path.` 

I also generally tried to use "this location" instead of "here".
I tried to formulate messages to use "depends on" and use the term "a user-provided value".

The previous change-note for changing the alert messages still cover this PR.

I'm also trying to [update the style-guide based on my findings](https://github.com/github/codeql/pull/10405).

### CODE REVIEW

**Comment by smowton** [comment]
Mostly looks good, though I'm personally not keen on `at this location` or `to this location`, because what is a location in this context? Generally speaking we're actually saying "is passed to a logging call on this line" or similar, where neither that call nor that argument wouldn't normally be described as a "location". Given the choice between a tangly and verbose expression of what's actually happening or the imprecise "here", I prefer the latter.

File: go/ql/src/Security/CWE-601/OpenUrlRedirect.ql [hunk 1]
```diff
@@ -24,5 +24,5 @@
   // this excludes flow from safe parts of request URLs, for example the full URL when the
   // doing a redirect from `http://<path>` to `https://<path>`
   not scfg.hasFlow(_, sink.getNode())
-select sink.getNode(), source, sink, "Untrusted URL redirection due to $@.", source.getNode(),
+select sink.getNode(), source, sink, "Untrusted URL redirection depends on a $@.", source.getNode(),
```
[the remark this answers is not in the document]
Comment (erik-krogh) [reply go/ql/src/Security/CWE-601/OpenUrlRedirect.ql:27]: A false positive. It's a data-flow query, but the configuration includes basically all the taint-tracking steps.

**Review by smowton — APPROVED** [review]

**Comment by smowton** [comment]
Does this want a change note since it might lead to alerts being considered spuriously resolved and recreated?

**Comment by erik-krogh** [comment]
> Does this want a change note since it might lead to alerts being considered spuriously resolved and recreated?

There is still [the change-note from my previous Go PR that touched alert-messages](https://github.com/github/codeql/blob/9979fa3f36629b4e689ba7f8bd8ecde37a94a1c3/go/ql/src/change-notes/2022-08-23-alert-messages.md). 
So that is covered.

**Pull request merged**